docs/product/specs/forms.md

```diff
@@ -0,0 +1,466 @@
+- The label displayed in the search bar & the rows would be from `lookup.columns[].label`.
+- Sorting would be in the ascending order for each column in `lookup.columns`.
+- Searching would be `starts_with` for text columns and `equals` for non-text columns. The combination between column-values would be `AND`.
+
+## Unauthenticated RPC actions
+- `forms.get` as mentioned in the initial iteration would be renamed to `forms.show`, which would not contain sensitive values like `owner_oid`.
+- We should have a separate `forms.get` which would be the private form definition that is fetched by Mathesar users.
+- The new UX would require a `forms.list_linked_records`, which will return the records & summaries for the list displayed to the user in the dropdown.
+
+## Django DB modal
+- Brent wants to spend some time thinking if `form.elements` should be a table instead of `jsonb`.
+- Pavish is fine with either, but prefers the `jsonb` since the actual performance consideration would be in creating the queries, not in parsing the json/reading from table.
+- This would not affect the API request/response structure.
+- This is an implementation detail and Brent would take a call during backend implementation work.
+
+# 2025-04-29: Initial iteration
+
+## Terminology
+- Reverse foreign key: Denotes foreign key columns in other tables that link to base table records.
+- Mathesar user: Refers to users using Mathesar to create a form.
+- End user: Refers to people filling the form.
+
+## Requirements
+
+### User stories
+
+- **Form Builder (Mathesar admin)**
+	- I want to share a form to end users via a public link to collect data required for my business needs.
+	- I want to select a base table & related tables, and choose which columns appear on my form so I can tailor data-entry to business needs.
+	- I want to ensure that all fields related to my business needs are part of the same form. This may involve multiple different tables related via FKs and reverse-FKs.
+	- I want end users to be able to search through existing records with summaries, when they're filling in a foriegn key field.
+	- I want to allow end users to be able to create new related records inline, if needed.
+	- I want to configure labels, descriptions, placeholders, and validation rules for each field to guide end users.
+	- I want to reorder fields via drag-and-drop and insert dividers or instructional text to structure my form visually.
+	- I want the main record and any related/nested records to all be created in one shot, when the end user submits the form.
+
+- **Form Submitter (End User)**  
+	- I want a clear, labelled form that only shows fields relevant to my task.
+	- I want descriptive messages to guide me fill the form.
+	- I want clear indications when the form is successfully submitted.
+	- I want useful help messages when form submission fails due to an error.
+	- I want to know when I'm filling the form incorrectly while I'm filling it, not only while submitting.
```

Comment:
This section isn't a list of user stories at all, it's a list of features prefixed with "I want". Attempting to wedge a list of feature requirements into a user story framework is more distracting than helpful, so I suggest dropping the pretext and reformatting/rewriting this section to be more readable, e.g., drop the "I want" prefixes, and consider sub-bulletpoints.

Reply:
@mathemancer I'm not sure we align on your definition on what an user story is. From what I've worked on in the past, user stories are usually feature descriptions written with "I want". Perhaps you're conflating user stories to use-case scenarios, which is presented after this section.

Here are a few documentations on user stories:
- https://www.mountaingoatsoftware.com/agile/user-stories
- https://www.productplan.com/glossary/user-story/